internal/v2/certverifier/certverifier_test.go

```diff
@@ -155,40 +163,136 @@ func TestVerifyClientCertChain(t *testing.T) {
+		},
+		{
+			description:               "chain of length 2 correct",
+			hostname:                  "host",
+			rawCerts:                  [][]byte{serverLeafDERCert, serverIntermediateDERCert},
+			expectedErr:               nil,
+			serverAuthorizationPolicy: []byte(fakeServerAuthorizationPolicy),
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			// Create new stream to S2Av2.
+			opts := []grpc.DialOption{
+				grpc.WithTransportCredentials(insecure.NewCredentials()),
+				grpc.WithReturnConnectionError(),
+				grpc.WithBlock(),
+			}
+			conn, err := grpc.Dial(address, opts...)
+			if err != nil {
+				t.Fatalf("Client: failed to connect: %v", err)
+			}
+			defer conn.Close()
+			c := s2av2pb.NewS2AServiceClient(conn)
+			log.Printf("Client: connected to: %s", address)
+			ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+			defer cancel()
+
+			// Setup bidrectional streaming session.
+			callOpts := []grpc.CallOption{}
+			cstream, err := c.SetUpSession(ctx, callOpts...)
+			if err != nil {
+				t.Fatalf("Client: failed to setup bidirectional streaming RPC session: %v", err)
+			}
+			log.Printf("Client: set up bidirectional streaming RPC session.")
+
+			// TODO(rmehta19): Add verificationMode to struct, and vary between tests.
+			VerifyPeerCertificateFunc := VerifyServerCertificateChain(tc.hostname, s2av2pb.ValidatePeerCertificateChainReq_CONNECT_TO_GOOGLE, cstream, tc.serverAuthorizationPolicy)
+			got, want := VerifyPeerCertificateFunc(tc.rawCerts, nil), tc.expectedErr
+			if want == nil {
+				if got != nil {
+					t.Errorf("Peer certificate verification failed, got: %v, want: %v", got, want)
+				}
+			} else {
+				if got == nil {
+					t.Errorf("Peer certificate verification failed, got: %v, want: %v", got, want)
+				}
+				if got.Error() != want.Error() {
+					t.Errorf("Peer certificate verification failed, got: %v, want: %v", got, want)
+				}
+			}
+		})
+	}
+	stop()
+}
+
+// TestVerifyServerCertChainWithoutServerAuthorizationPolicy runs unit tests for VerifyServerCertificateChain without server authorization policy check.
+func TestVerifyServerCertChainWithoutServerAuthorizationPolicy(t *testing.T) {
+	// Start up fake S2Av2 server.
+	var wg sync.WaitGroup
+	wg.Add(1)
+	stop, address, err := startFakeS2Av2Server(&wg, false)
+	wg.Wait()
+	if err != nil {
+		t.Fatalf("Error starting fake S2Av2 Server: %v", err)
+	}
+
+	for _, tc := range []struct {
+		description               string
+		hostname                  string
+		rawCerts                  [][]byte
+		expectedErr               error
+		serverAuthorizationPolicy []byte
+	}{
+		{
+			description:               "empty chain",
+			hostname:                  "host",
+			rawCerts:                  nil,
+			expectedErr:               errors.New("server cert verification failed: server peer verification failed: server cert chain is empty"),
+			serverAuthorizationPolicy: []byte(fakeServerAuthorizationPolicy),
```

Comment:
probably should be nil? since the tests are for without server auth policy

Reply:
Sorry for the confusion, I think the "without server auth policy check" means the fake S2A server won't check the server authz policy field, hence, no matter what value you set to the field, the test should still pass.